app/src/main/java/com/example/amcc/view/MainFragment.java

```diff
@@ -54,47 +54,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        viewModel = new ViewModelProvider(requireActivity()).get(ShardViewModel.class);
+        viewModel = new ViewModelProvider(requireActivity()).get(SharedViewModel.class);
 
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        validation = new AwesomeValidation(ValidationStyle.BASIC);
-
-        spinner = view.findViewById(R.id.spinnerCityHolderID);
-        emissionEdtField = view.findViewById(R.id.edtNumEmissionID);
-        engineSizeField = view.findViewById(R.id.edtNumEngSizeID);
-        avgConField = view.findViewById(R.id.edtNumConsumeID);
-        milePerYField = view.findViewById(R.id.edtNumMileAgeYearID);
-//        calendarView = view.findViewById(R.id.calendarViewID);
-//        germanFormat = new SimpleDateFormat("dd.MM.yyyy", Locale.GERMANY);
-        //       regDate = germanFormat.format(calendarView.getDate());
-        //     calendarView.setOnDateChangeListener((view1, year, month, dayOfMonth) -> regDate = "" + dayOfMonth + "." + (month + 1) + "." + year);
-        regDateField = view.findViewById(R.id.edtFirst_reg_year);
 
+        navController = Navigation.findNavController(view);
+        validation = new AwesomeValidation(ValidationStyle.BASIC);
         //Create City Name List
         setUpCityNamesList();
 
-        navController = Navigation.findNavController(view);
         view.findViewById(R.id.btnResultID).setOnClickListener(v -> {
+            emissionEdtField = view.findViewById(R.id.edtNumEmissionID);
+            engineSizeField = view.findViewById(R.id.edtNumEngSizeID);
+            avgConField = view.findViewById(R.id.edtNumConsumeID);
+            milePerYField = view.findViewById(R.id.edtNumMileAgeYearID);
+            regDateField = view.findViewById(R.id.edtFirst_reg_year);
+
+            radioGroup = view.findViewById(R.id.fuel_type_radio_group);
+            int radioId = radioGroup.getCheckedRadioButtonId();
+            radioButton = view.findViewById(radioId);
+            String fuelType = radioButton.getText().toString();
```

Comment:
@Jaber-aldroubi Here you have big mistake! What value does fuelType return?

Reply:
fuelType return the string, which the user choose (E10, Super, Diesel)

Reply:
Thy will be later converted to e10, e5, diesel. Case letter is also important